app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -535,6 +535,22 @@ private void switchToImmersiveMode() {
                         | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
     }
 
+    @Override
+    public void onDestroy() {
```

Comment:
nit: For correctness, I think we should set a boolean, `isImmersive`, when entering/exiting immersive mode and deactivate/activate as necessary in `onStart`/`onStop`.

I'm concerned that `onDestroy` is not guaranteed to be called. Sensically, the app isn't being closed so the system should clean up the resources when transitioning between fragments but since it's not contractually obligated to call `onDestroy`, I'm concerned it won't, especially on obscure devices with modified Android.

That being said, if you disagree and stick with onDestroy:
- I think you should add a comment (maybe above `exitImmersive...`) about why we don't care that `onDestroy` isn't guaranteed to be called
- Consider using `onDetach` - wouldn't it be more correct? According [to the lifecycle](https://cdn-images-1.medium.com/max/1600/0*YWCr1M1pwWfvCzvZ.png), there are times when `onDetach` will be called but `onDestroy` won't be. It's also documented [in the `setRetainInstance` javadoc](https://developer.android.com/reference/android/app/Fragment.html#setRetainInstance(boolean)).

Reply:
> onStart/onStop would mean we leave immersive mode if the app is not visible anymore. However right now it's perfectly fine to leave the app and then get back to your full screen video.

True. I guess the flag we really want to clear onStart/onStop is the SCREEN_ON one, which breaks the abstraction we have around immersiveMode but there's probably a way to clean it up (e.g. an object we delegate onStart/onStop to).

> This is not a fragment that retains the instance (You have to manually call setRetainInstance yourself for that).

I was just future proofing. :)

Reply:
Android will take care of that. No need to manually remove the flag when switching apps etc.: https://developer.android.com/training/scheduling/wakelock.html